spec.html

```diff
@@ -36557,27 +36557,73 @@
       <p>If an ECMAScript implementation has a mechanism for reporting diagnostic warning messages, a warning should be produced when code contains a |FunctionDeclaration| for which these compatibility semantics are applied and introduce observable differences from non-compatibility semantics. For example, if a var binding is not introduced because its introduction would create an early error, a warning message should not be produced.</p>
+      <emu-annex id="sec-web-compat-block-static-semantics-lexicallydeclarednames">
+        <h1>Changes to Block LexicallyDeclaredNames</h1>
+        <p>In Static Semantics: LexicallyDeclaredNames of <emu-grammar>StatementList : StatementList StatementListItem</emu-grammar> (<emu-xref href="#sec-block-static-semantics-lexicallydeclarednames"></emu-xref>) the following is performed in place of step 2:</p>
+        <emu-alg>
+          1. If the production is contained in strict mode code, then
+            1. Append to _names_ the elements of the TopLevelLexicallyDeclaredNames of |StatementListItem|.
+          1. Else,
+            1. If |StatementListItem| is <emu-grammar>StatementListItem : Declaration</emu-grammar> and |Declaration| is <emu-grammar>Declaration : HoistableDeclaration</emu-grammar> and |HoistableDeclaration| is <emu-grammar>HoistableDeclaration : FunctionDeclaration</emu-grammar> , then
+              1. Let _fn_ be the sole element of the BoundNames of |FunctionDeclaration|.
+              1. If _names_ does not include the element _fn_, append _fn_ to _names_.
+            1. Else,
+              1. Append to _names_ the elements of the TopLevelLexicallyDeclaredNames of |StatementListItem|.
+        </emu-alg>
+      <emu-note>
+        Multiple |FunctionDeclaration|s for the same name inside a block is allowed when not in strict mode code.
+      </emu-note>
+      </emu-annex>
+      <emu-annex id="sec-web-compat-blockdeclarationinstantiation">
+        <h1>Changes to BlockDeclarationInstantiation</h1>
+        <p>During BlockDeclarationInstantiation (<emu-xref href="#sec-blockdeclarationinstantiation"></emu-xref>) the following steps are performed in place of step 4:</p>
+        <emu-alg>
+          1. Let _functionsToInitialize_ be an empty List.
+          1. For each _d_ in _declarations_, in reverse list order do
+            1. For each element _dn_ of the BoundNames of _d_ do
+              1. If IsConstantDeclaration of _d_ is *true*, then
+                1. Perform ! _envRec_.CreateImmutableBinding(_dn_, *true*).
+              1. Else,
+                1. Perform ! _envRec_.CreateMutableBinding(_dn_, *false*).
+            1. If _d_ is a |GeneratorDeclaration| production, then
+              1. Insert _d_ as the first element of _functionsToInitialize_.
+            1. If _d_ is a |FunctionDeclaration| production, then
+              1. If _d_ is not an element of _functionToInitialize_, then
```

Comment:
This condition is always true, you need to add a separate list of function names to keep track of duplicates.